db/migrate/20230726100204_add_parent_ids.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class AddParentIds < ActiveRecord::Migration[7.0]
+  def change
+    add_column :model_solutions, :parent_id, :integer, null: true, limit: 1
+    add_column :tests, :parent_id, :integer, null: true, limit: 1
+    add_column :task_files, :parent_id, :integer, null: true, limit: 1
```

Comment:
Shouldn't this be rather a foreign key constrain (with `null: true`) rather than just a single integer column?

Reply:
It is now added.

Reply:
Where did you add these? I would have expected to see a change in `db/migrate/20230726100204_add_parent_ids.rb`, but cannot see it there.

Reply:
I have added a separate migration converting the column to a foreign key.
It can be found here `db/migrate/20240513071837_parent_id_to_foreign_key.rb`.

Reply:
Thank you 🙏 

Using a dedicated migration is fine. Still, we could have used the very same migration as well, since the changes are not yet merged upstream.

Regarding the actual migration, I continue to have a small change request: The data type..
- With your first migration, you specify the data type as `integer` manually and with your second migration you specify the foreign key constraint. When looking at the schema, the original table IDs use the `:serial` format, or specifically are `:bitint`. Now, those two data types are mixed and it could happen that a numerically large ID cannot be stored as a `parent`. 
- The issue might even occur faster than with large numbers, since the option `limit: 1` was used in your original migration. `:limit` specifies the "number of bytes" used for this column (the minimum, can be potentially larger I assume). With 1 byte being 8 bits, we have a maximum of 2^8 = 256. I haven't tested, but are we sure that ` parent_id`s >= 256 can be stored?